felix/labelindex/named_port_index.go

```diff
@@ -316,23 +316,27 @@ type SelectorAndNamedPortIndex struct {
 	lastLiveReport time.Time
 }
 
-func NewSelectorAndNamedPortIndex() *SelectorAndNamedPortIndex {
+func NewSelectorAndNamedPortIndex(deduplicate bool) *SelectorAndNamedPortIndex {
```

Comment:
I might call this `suppressOverlaps` just to avoid confusion with deduplicating _exact_ duplicate entries, which it already does (IIRC)